tests-integration/tests/crowdfund_app.rs

```diff
@@ -0,0 +1,573 @@
+
+struct TestCase {
+    router: MockApp,
+    andr: MockAndromeda,
+    crowdfund: MockCrowdfund,
+    cw20: Option<MockCW20>,
+    cw721: MockCW721,
+    presale: Vec<PresaleTierOrder>,
+}
+
+#[fixture]
+fn setup(#[default(true)] use_native_token: bool) -> TestCase {
+    let mut router = mock_app(None);
+    let andr = MockAndromedaBuilder::new(&mut router, "admin")
+        .with_wallets(vec![
+            ("owner", vec![]),
+            ("buyer_one", vec![coin(1000000, "uandr")]),
+            ("recipient", vec![]),
+        ])
+        .with_contracts(vec![
+            ("cw20", mock_andromeda_cw20()),
+            ("cw721", mock_andromeda_cw721()),
+            ("crowdfund", mock_andromeda_crowdfund()),
+            ("app-contract", mock_andromeda_app()),
+        ])
+        .build(&mut router);
+
+    let owner = andr.get_wallet("owner");
+    let buyer_one = andr.get_wallet("buyer_one");
+    let recipient = Recipient::new(andr.get_wallet("recipient"), None);
+
+    // Add cw721 component
+    let cw721_init_msg = mock_cw721_instantiate_msg(
+        "Campaign Tier".to_string(),
+        "CT".to_string(),
+        "./crowdfund".to_string(),
+        None,
+        andr.kernel.addr().to_string(),
+        None,
+    );
+
+    let cw721_component = AppComponent::new(
+        "cw721".to_string(),
+        "cw721".to_string(),
+        to_json_binary(&cw721_init_msg).unwrap(),
+    );
+
+    let mut app_components = vec![cw721_component.clone()];
+
+    // Add cw20 components for test cases using cw20
+    let cw20_component: Option<AppComponent> = match use_native_token {
+        true => None,
+        false => {
+            let initial_balances = vec![Cw20Coin {
+                address: buyer_one.to_string(),
+                amount: Uint128::from(1000000u128),
+            }];
+            let cw20_init_msg = mock_cw20_instantiate_msg(
+                None,
+                "Test Tokens".to_string(),
+                "TTT".to_string(),
+                6,
+                initial_balances,
+                Some(mock_minter(
+                    owner.to_string(),
+                    Some(Uint128::from(1000000u128)),
+                )),
+                None,
+                andr.kernel.addr().to_string(),
+            );
+            let cw20_component = AppComponent::new(
+                "cw20".to_string(),
+                "cw20".to_string(),
+                to_json_binary(&cw20_init_msg).unwrap(),
+            );
+            app_components.push(cw20_component.clone());
+            Some(cw20_component)
+        }
+    };
+    let denom = match use_native_token {
+        true => Asset::NativeToken("uandr".to_string()),
+        false => Asset::Cw20Token(AndrAddr::from_string(format!(
+            "./{}",
+            cw20_component.clone().unwrap().name
+        ))),
+    };
+    // Add campaign component
+    // *** IMPORTANT ***
+    //    campaign component should be added in the last order
```

Comment:
Ordering shouldn't matter thanks to `Instantiate2`

Reply:
Hmm. Actually tried by adding cw20 at the end of the `app_components` and it threw error `InvalidAsset` error with `./cw20`

Reply:
Oh hmmm, I wonder if its cause it's querying the contract for cw20 info and it doesn't exist yet

Probably good to alert the FE of this unless we can come up with a good solution

CC: @SlayerAnsh